pyccel/parser/semantic.py

```diff
@@ -1087,23 +1092,23 @@ def incompatible(i_arg, f_arg):
                     or f_arg.dtype is NativeGeneric()):
                 continue
 
-            err_msgs = []
             # Check for compatibility
             if incompatible(i_arg, f_arg):
                 expected  = self.get_type_description(f_arg, not elemental)
                 type_name = self.get_type_description(i_arg, not elemental)
                 received  = f'{i_arg} ({type_name})'
-                err_msgs += [INCOMPATIBLE_ARGUMENT.format(idx+1, received, expr, expected)]
+                err_msgs += [INCOMPATIBLE_ARGUMENT.format(idx+1, received, func_name, expected)]
 
             if f_arg.rank > 1 and i_arg.order != f_arg.order:
-                err_msgs += [INCOMPATIBLE_ORDERING.format(idx=idx+1, arg=i_arg, func=expr, order=f_arg.order)]
+                err_msgs += [INCOMPATIBLE_ORDERING.format(idx=idx+1, arg=i_arg, func=func_name, order=f_arg.order)]
 
-            if err_msgs:
-                flag = False
-                if raise_error:
-                    errors.report('\n'.join(err_msgs), symbol = expr, severity='fatal')
-                else:
-                    return flag
+        if err_msgs:
+            flag = False
+            if raise_error:
+                bounding_box=(self.current_ast_node.lineno, self.current_ast_node.col_offset)
+                errors.report('\n\n'.join(err_msgs), symbol = expr, bounding_box=bounding_box, severity=error_type)
```

Comment:
You don't need a bounding box if the `expr` is a `PyccelAstNode`. With the introduction of `func_name` this is the case.
```suggestion
errors.report('\n\n'.join(err_msgs), symbol = expr, severity=error_type)
```

Reply:
The issue is that sometimes I don't have the `FunctionCall` node, that is why I used this because I know that we call this function only when we are in a `FunctionCall`, so I don't need to pass it, otherwise I will have to pass it to `_visit_FunctionDef`, I tried to see if it is possible to raise the error elsewhere, but it was not possible

Reply:
Ah ok. I thought `expr` was a `FunctionDef` or a `FunctionCall`. In that case the docstring for this function needs updating

Reply:
I can still pass the FunctionDef, but the line will be incorrect, we want to error to point the line of the FunctionCall

Reply:
Ok. Then I guess you just need to fix the docstring

Reply:
I think we just need to pass the FunctionDef always so that we have a unified way of using the function

Reply:
[8707e5e](https://github.com/pyccel/pyccel/pull/1728/commits/8707e5e515190f20099d19bc59f6ff80b5375b87)